common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowTask.java

```diff
@@ -533,6 +533,9 @@ public TaskDef getTaskDefinition() {
      * @param taskDefinition Task definition
      */
     public void setTaskDefinition(TaskDef taskDefinition) {
+        if (taskDefinition != null && taskDefinition.getName() == null) {
+            taskDefinition.setName(this.name);
```

Comment:
I would not expect a `setX(X arg)` to mutate the passed parameter. It’s not ideal to introduce side effects to a passed parameter. 

Either avoid the side effect, document it clearly or create a copy.

Reply:
@shaileshpadave can you check where is this being called from and the caller should fix this.